src/main.ml

```diff
@@ -52,6 +52,25 @@ let msat =
     pre; solve;
   }
 
+(* mc² *)
+let mc2 =
+  let open Mc2_core in
+  let pre clauses =
+    let solver = Solver.create ~plugins:[Mc2_dimacs.Plugin_sat.plugin] () in
```

Comment:
I'd prefer the solver to be create in the `solve` function, as id done for the other sat solvers used, or is it really not possible ?

Reply:
This is not possible. I cannot create terms without the solver (they depend on internal plugin IDs).